api/controllers/ViewController.js

```diff
@@ -95,7 +95,9 @@ module.exports = {
     if (request.query.client_id) {
       try {
         const client = await Client.findOne({ id: request.query.client_id });
-        if (!client || (client && client.redirectUri !== request.query.redirect_uri)) {
+        if (!client
+          || (client && client.redirectUri !== request.query.redirect_uri
+            && client.redirectUrls.indexOf(request.query.redirect_uri) === -1)) {
```

Comment:
`client.redirectUrls.includes()` would read better to me, but this code is of course valid.

Reply:
Agree. I changed it.